core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -209,13 +210,14 @@ abstract class RDD[T: ClassTag](
    */
   def cache(): this.type = persist()
 
+  private val unpersistBlocking = conf.get(RDD_UNPERSIST_BLOCKING)
   /**
    * Mark the RDD as non-persistent, and remove all blocks for it from memory and disk.
    *
    * @param blocking Whether to block until all blocks are deleted.
    * @return This RDD.
    */
-  def unpersist(blocking: Boolean = true): this.type = {
+  def unpersist(blocking: Boolean = unpersistBlocking): this.type = {
```

Comment:
What is difference between calling with explicit parameter and controlling by a config?

Reply:
@viirya In a production environment, the configs and packages are always managed by gray release systems, so a config is needed rather than the explicit parameter.
If you agree, it would be better to make the blocking parameter default to `false` as @srowen suggested, and it's what I expected. I proposed to make this blocking parameter configurable is afraid of the side effect of this changes.

Reply:
As @srowen said, I'd say that it is controlled by the caller. You can decide to call it with blocking=false without using default value.

I'm not sure if making false as default will affect existing programs. After a call to unpersist without blocking parameter, the blocks may still exist, will it cause unexpected effect? Like OOM?

Reply:
@viirya 

> I'd say that it is controlled by the caller. You can decide to call it with blocking=false without using default value.
Yes, I agree it's controlled by the caller, what we change is only the default value, callers can always set the blocking parameter by themselves. 

> I'm not sure if making false as default will affect existing programs. After a call to unpersist without blocking parameter, the blocks may still exist, will it cause unexpected effect? Like OOM?
Actually, the blocking just wait until RpcTimeout or throws exception if error occurred. we still can not expect the blocks being cleaned up, and users may be confusing if an exception is thrown, because in most cases, users don't known how to handle this. What's more, users always use rdd.unpersist as non-exceptional(no explicit warnings or errors).

I agree with @srowen to make it default to `false`, since it should be rare that someone wants to wait for it to unpersist.

I have also checked the commit history, seems there is not special intention for this default value.
https://github.com/apache/spark/blob/ba5e544461e8ca9216af703033f6b0de6dbc56ec/core/src/main/scala/spark/RDD.scala

Reply:
Maybe this pr is a temp solution for how to avoid RPC issue. There are unable to prove the the measure is effective before a lot of discussion about the blocking of unpersist in detail.